### PR TITLE
ref(releases): Function to handle resolved in release

### PR DESCRIPTION
**Pull request by armenzg**
This is the largest code block in update_groups. This moves all related logic into a single function.

For the record, I am not focusing on writing better code but on making it more encapsulated. Proper refactors could happen later.

### CODE REVIEW

File: src/sentry/api/helpers/group_index/update.py [hunk 1]
```diff
@@ -216,13 +216,7 @@ def update_groups(
     project_lookup = {p.id: p for p in projects}
 
     acting_user = user if user.is_authenticated else None
-    self_assign_issue = "0"
```
Comment (armenzg) [inline src/sentry/api/helpers/group_index/update.py:219]: All of these lines are only used within the new function.

File: src/sentry/api/helpers/group_index/update.py [hunk 2]
```diff
@@ -264,305 +254,18 @@ def update_groups(
             project_lookup=project_lookup,
         )
     if status in ("resolved", "resolvedInNextRelease"):
-        res_status = None
```
Comment (armenzg) [inline src/sentry/api/helpers/group_index/update.py:267]: All code in here has been moved into the function with minimal (I will point it out if I do) or no code changes.

File: src/sentry/api/helpers/group_index/update.py [hunk 3]
```diff
@@ -591,7 +294,7 @@ def update_groups(
             else:
                 result["statusDetails"] = {}
         if group_list and status_updated:
-            activity_type, activity_data = handle_status_update(
+            handle_status_update(
```
Comment (armenzg) [inline src/sentry/api/helpers/group_index/update.py:297]: I analyzed the code from a couple of weeks ago, making sure I did not regress it. However, `activity_type` and `activity_data` are not used from this point on in the code:


https://github.com/getsentry/sentry/blob/12d04478fab99de1787df334ce77d36632fda07d/src/sentry/api/helpers/group_index/update.py#L610-L621
https://github.com/getsentry/sentry/blob/58657d1060b3e67a76b40867ca7474aa7b0bd19b/src/sentry/issues/status_change.py#L144-L150

It seems that activity is read back from the DB here, thus, no need to return it as variables:
https://github.com/getsentry/sentry/blob/12d04478fab99de1787df334ce77d36632fda07d/src/sentry/api/helpers/group_index/update.py#L632-L637

File: src/sentry/api/helpers/group_index/update.py [hunk 4]
```diff
@@ -633,6 +336,314 @@ def update_groups(
...
+def handle_resolve_in_release(
+    status: str,
+    status_details: Mapping[str, Any],
+    group_list: Sequence[Group],
+    projects: Sequence[Project],
+    project_lookup: Mapping[int, Project],
+    acting_user: User | None,
+    user: User | RpcUser,
+    result: MutableMapping[str, Any],
+) -> tuple[dict[str, Any], GroupResolution.Type | None] | Response:
```
Comment (armenzg) [inline src/sentry/api/helpers/group_index/update.py:348]: Notice that we either return data or a response. This is handled at the caller level:
image

File: src/sentry/api/helpers/group_index/update.py [hunk 5]
```diff
@@ -633,6 +336,314 @@ def update_groups(
...
+    res_type = None
+    release = None
+    commit = None
+    self_assign_issue = "0"
+    if acting_user:
+        user_options = user_option_service.get_many(
+            filter={"user_ids": [acting_user.id], "keys": ["self_assign_issue"]}
+        )
+        if user_options:
+            self_assign_issue = user_options[0].value
```
Comment (armenzg) [inline src/sentry/api/helpers/group_index/update.py:358]: The lines above are the only ones I'm adding to the original conditional block code.
image

All following code is as it was written originally. Later this week I will break up this function into a few more functions and I will be satisfied for now.

File: src/sentry/issues/status_change.py [hunk 6]
```diff
@@ -74,12 +73,10 @@ def handle_status_update(
     status_details: dict[str, Any],
     acting_user: User | None,
     sender: Any,
-) -> ActivityInfo:
+) -> None:
```
Comment (armenzg) [inline src/sentry/issues/status_change.py:76]: No code in the codebase makes use of the returned value.

**Review by snigdhas — APPROVED** [review]

**Pull request merged**